config/initializers/node-orm2-connect.js

```diff
@@ -15,9 +15,5 @@ export default {
       application.logger.error('Error initializing the node-orm2 adapter or database connection:');
       application.logger.error(error.stack);
     }
-
-    if (config.syncSchema) {
-      await fromNode((cb) => adapter.db.sync(cb));
-    }
```

Comment:
In other words, the `define-orm-models` initializer (from denali itself) needs to know about the database connection in order to define the models, but the schema sync function needs to have all the models defined before it is executed (otherwise it doesn't know about the schema that it needs to sync). So basically, we have to split it into two initializers for the syncSchema functionality to work properly.